Scripts/Editor/Utils/CodeGenerationUtility.cs

```diff
@@ -182,27 +183,55 @@ public static void AppendFooter(StreamWriter writer, ref int indentation, string
             }
         }
 
+        public static void ValidateIfCanBePartial(ScriptableObjectCollection collection)
```

Comment:
Maybe this is a matter of preference/what you're used to but if I saw a method called `ValidateIfCanBePartial` I wouldn't expect it to modify the collection, I'd expect it to return a boolean or something.

If you wanted to be more explicit about what it does, you could consider renaming it to something like `DisablePartialClassGenerationIfDisallowed`